developers/release.md

```diff
@@ -0,0 +1,240 @@
+
+```
+cd katello
+git fetch origin
+git checkout master
+git reset origin/master --hard
+tito tag
+cd deploy/
+tito tag
+git checkout -b KATELLO-X.Y
+```
+
+Update the releasers and tito configuration for each package (some repositories contain multiple packages, see Step 3 for list):
+
+```
+vim rel-eng/releasers.conf
+:tabnew rel-eng/tito.props
+
+:tabdo %s/katello-nightly-rhel6/katello-X.Y-rhel6/g
+:tabdo %s/katello-nightly-rhel7/katello-X.Y-rhel7/g
+:wqa
+
+git commit -a -m 'Updating rel-eng for X.Y'
+git push origin KATELLO-X.Y
+```
+
+Now we set master to start from the version after the one we are working on releasing X.Y+1, for each package, ensuring you are in the directory the spec file exists:
+
+```
+git checkout master
+```
+
+For any of the gems, update the gem version file and the spec file to X.Y.0 and make sure to reset the release to 1.
+
+```
+vim lib/katello/version.rb
+vim rubygem-katello.spec
+git commit -a -m 'Version bump to X.Y.0'
+```
+
+Now for all of the repositories including the gems, tag and push the starting place for the next version:
+
+```
+tito tag --use-version=X.Y+1.0
+git push origin master && git push --tags
+```
+
+For `hammer-cli-katello`:
+
+1. If there are no changes since last version do nothing
+2. If there are changes, bump the version, release the gem, tag and build in koji
+
+Lastly, branch documentation
+
+Now we are ready to branch the documentation. This is done separately from the other branching due to the fact that we do not tag or generate an RPM of the documentation website.
+
+```
+git clone https://github.com/Katello/katello.org
+cd katello.org
+git checkout -b KATELLO-X.Y
+git push origin KATELLO-X.Y
+```
+
+Check that the CI job that builds the site (http://ci.theforeman.org/job/deploy_katello_site/) has been kicked off or kick it off manually if needed. Once it passes, visit [http://www.katello.org](http://www.katello.org) and check that X.Y is available from the drop down list. If it did not build correctly, you will need to checkout the `deploy` branch and test it locally.
+
+Congratulations, you have successfully branched all of the repositories and prepared them for release. We recommend double checking each repository ensuring that the branch exists in the upstream repository, that the top of the commit stack in that branch includes the tag you created prior to branching (so that you have an initial tag point that represents the branch point) and that the tag itself exists. The last part of this step is to send to a follow up email letting the community know that everything has been branched.
+
+### Step 5: Configure Koji
```

Comment:
Need a step here:

### Step 5. Pin katello-installer puppet modules for release

On the katello-installer's KATELLO-X.Y branch, we need to pin the modules to a minor release.

First, we take the pinned Foreman modules in the Puppetfile from the most current release, e.g. https://github.com/theforeman/foreman-installer/blob/1.9-stable/Puppetfile.  Overwrite the Foreman section in katello-installer's Puppetfile with these locked down versions.

Next, we'll lock down the Katello modules. First, Check out all the puppet modules:

```bash
for module in candlepin certs capsule common elasticsearch gutterball katello crane service_wait pulp
do
  git clone git@github.com:Katello/puppet-${module}.git
done
```

Review the modules for releasing - we should be keeping them up to date when making incompatible changes, but do any need bumps of the major or minor version? If so, do so now.

Consider the changes that have been made to the module since the last release, do we need a new release?
Use [blacksmith](https://github.com/maestrodev/puppet-blacksmith) to push to forge.

```bash
rake module:release
```

Now, lock down Katello versions to the appropriate minor, e.g. if you release `puppet-katello 0.2.0` or the latest is 0.2.0, then in the Puppetfile

```
mod 'katello',   '>= 0.2.0 < 0.3.0'
```